spark/src/test/scala/org/apache/spark/sql/delta/test/CustomCatalogs.scala

```diff
@@ -138,6 +138,76 @@ class DummySessionCatalogInner extends DelegatingCatalogExtension {
   }
 }
 
+/**
+ * A session-catalog wrapper used in tests to simulate Unity Catalog managed tables by injecting
+ * UC-related properties into the returned [[CatalogTable.storage.properties]].
+ *
+ * This allows end-to-end plan construction (and streaming query execution attempts) to observe
+ * UC-managed markers through Spark's normal catalog resolution paths, without manually patching
+ * CatalogTable objects in tests.
+ */
+class UCTableInjectingSessionCatalogInner extends DelegatingCatalogExtension {
```

Comment:
why extending `DelegatingCatalogExtension`? customizing the `loadTable` method of UCTableInjectingSessionCatalog seems enough